### PR TITLE
SN1-93 - Synapse query

### PR DESCRIPTION
**Pull request by nnboskovic**
Added an implementation for running a read cypher query on a synapse

@aph5nt @0xMarx Let me know what the best way to test this would be

### CODE REVIEW

File: neurons/miners/bitcoin/funds_flow/graph_search.py [hunk 1]
```diff
@@ -32,9 +34,14 @@ def __init__(
             auth=(self.graph_db_user, self.graph_db_password),
         )
 
-    def execute_query(self, network, query):
-        # TODO: Implement this
-        return []
+    def execute_cypher_query(self, query):
```
Comment (0xMarx) [inline neurons/miners/bitcoin/funds_flow/graph_search.py:37]: does this work? it should be call with mandatory `terms` parameter no?

File: neurons/miners/miner.py [hunk 2]
```diff
@@ -102,11 +102,15 @@ def __init__(self, config=None):
             forward_fn=self.query,
             blacklist_fn=self.query_blacklist,
             priority_fn=self.query_priority,
+        ).attach(
+            forward_fn=self.cypher_query,
```
Comment (0xMarx) [inline neurons/miners/miner.py:106]: the cypher query is the query endpoint, you should implement the `query()` function

File: neurons/miners/miner.py [hunk 3]
```diff
@@ -102,11 +102,15 @@ def __init__(self, config=None):
...
+            blacklist_fn=self.query_blacklist,
+            priority_fn=self.query_priority,
         ).attach(
             forward_fn=self.challenge,
             blacklist_fn=self.challenge_blacklist,
             priority_fn=self.challenge_priority,
-        )
+        ))
```
Comment (0xMarx) [inline neurons/miners/miner.py:113]: the parentheses () are not needed in this case

File: neurons/utils.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        patterns.append(''.join(mixed_case_variations))
+
+        # Detecting comments that might hide portions of malicious queries
+        # This is a simplistic approach and might need refinement
+        patterns.append(f'/{escaped_term}|{escaped_term}/')
+
+    return patterns
+
+
+def is_potentially_malicious(query, terms):
```
Comment (0xMarx) [inline neurons/utils.py:32]: is_malicious is better then is_potentially_malicious as a function name

File: neurons/miners/bitcoin/funds_flow/graph_search.py (hunk 1 above)
Comment (0xMarx) [reply neurons/miners/bitcoin/funds_flow/graph_search.py:37]: also `execute_query` should be the name as we won't execute any other type of query.

**Review by 0xMarx — CHANGES_REQUESTED** [review]

**Review by ronnie-samaroo — APPROVED** [review]
good enough for MVP

**Pull request merged**